pkg/controller.v1beta2/tensorflow/controller.go

```diff
@@ -430,6 +496,76 @@ func (tc *TFController) satisfiedExpectations(tfjob *tfv1beta2.TFJob) bool {
 	return satisfied
 }
 
+// pastBackoffLimitOnFailure checks if container restartCounts sum exceeds BackoffLimit
+// this method applies only to pods with restartPolicy == OnFailure
+func (tc *TFController) pastBackoffLimitOnFailure(tfjob *tfv1beta2.TFJob, pods []*v1.Pod) (bool, error) {
+	if tfjob.Spec.BackoffLimit == nil {
+		return false, nil
+	}
+
+	result := int32(0)
+	for rtype, spec := range tfjob.Spec.TFReplicaSpecs {
+		if spec.RestartPolicy != common.RestartPolicyOnFailure {
```

Comment:
What about exit code?

Reply:
I think I need some suggestions for exit code.
There will be no restart count in the pod itself.
One way is we store the restart counts of pods in the status when deleting the pods in ReconcilePod, but this involves changing the status api.

Reply:
I see the problem, we use a sort of hacky way to force the pod to restart on certain exit codes: https://github.com/kubeflow/tf-operator/blob/master/pkg/controller.v1beta2/tensorflow/pod.go#L112

I am ok with only supporting BackoffLimit for OnFailure restarts, maybe just clarify this in the documentation. @johnugeorge ?